website/blog/2025-04-08-react-native-0.79.md

```diff
@@ -0,0 +1,286 @@
+
+#import <Foundation/Foundation.h>
+#import <ReactCommon/RCTTurboModule.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface <YourNativeModule>Provider : NSObject <RCTModuleProvider>
+
+@end
+```
+
+```objc title=”CppNativeModuleProvider.mm”
+
+NS_ASSUME_NONNULL_END
+
+// .mm file
+#import "<YourNativeModule>Provider.h"
+#import <ReactCommon/CallInvoker.h>
+#import <ReactCommon/TurboModule.h>
+#import "<YourNativeModule>.h"
+
+@implementation NativeSampleModuleProvider
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
+    (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+  return std::make_shared<facebook::react::NativeSampleModule>(params.jsInvoker);
+}
+```
+
+</details>
+
+With this new approach, we unified the registration of Native Modules for both app developers and library maintainers. Libraries can specify the same properties in their `package.json` and Codegen will take care of the rest.
+
+This approach solves the limitation we introduced in 0.77 that prevented the registration of a pure C++ Native Module with a Swift AppDelegate. As you can see, none of these changes modifies the AppDelegates and the generated code will work for AppDelegates implemented with both Swift and Objective-C.
+
+### Android: Faster App Startup
+
+We’re also shipping a change to improve your Android startup time by a significant amount.
+
+Starting with this version, we won’t be compressing the JavaScript bundle anymore inside the APK. Previously, the Android system needed to uncompress the JavaScript bundle before your app could start. This was causing a significant slowdown during the app startup.
+
+Starting from this release, we will be shipping the Javascript Bundle uncompressed by default, so your Android apps will be generally faster to start.
+
+We tested this feature on the Discord app and got a significant performance boost:
+
+Discord’s time-to-interactive (TTI) was reduced by 400ms, which was a 12% speedup with a one-line change (tested on a Samsung A14).
+
+On the other hand, storing the bundle uncompressed, will result in a higher space consumption for your application on the user device. If this is a concern to you, you can toggle this behavior using the `enableBundleCompression` property in your `app/build.gradle` file.
+
+```kotlin
+react {
+  // ...
+  // If you want to compress the JS bundle (slower startup, less
+  // space consumption)
+  enableBundleCompression = true
+  // If don't you want to compress the JS bundle (faster startup,
+  // higher space consumption)
+  enableBundleCompression = false
+
+  // Default is `false`
+}
+```
+
+Please note that the APK size will increase in this release, but your users won’t be paying the extra cost in APK download size, as the APKs are compressed when downloaded from the network.
+
+### Android: Faster Builds with Configuration Caching
+
+:::info
+
+Please note that this is an advanced feature that requires understanding of Android Development and build tools.
+
+:::
```

Comment:
if this is an advanced feature... does it makes sense to have it as an highlight? 🤔 I think that we usually highlights only features that are useful to most of our users.
Shouldn't we just improve the [Speeding up your Build phase](https://reactnative.dev/docs/build-speed) section, instead?

Reply:
Yeah we can move it there and have a mention somewhere in the blogpost (don't know where though)

Reply:
I've moved it entireyl